### PR TITLE
#47 filters for agency open + direct

### PR DESCRIPTION
**Pull request by elyk**
- Filters added for open and direct selection
- ordering ability added
- added new field for creation date
- updated tests

This does *not cover* unsolicited as there needs to be an endpoint added for that.

### CODE REVIEW

File: backend/unpp_api/apps/project/filters.py [hunk 1]
```diff
@@ -12,10 +14,16 @@ class BaseProjectFilter(django_filters.FilterSet):
...
+    posted_from_date = DateFilter(name='created',
+                                  lookup_expr=('gt'))
+    posted_to_date = DateFilter(name='created',
+                                lookup_expr=('lt'))
+    selected_source = CharFilter(lookup_expr=('iexact'))
 
     class Meta:
         model = EOI
-        fields = ['title', 'country_code', 'locations', 'specializations', 'status']
+        fields = ['title', 'country_code', 'locations', 'specializations', 'agency', 'active', 'selected_source',]
```
Comment (lkolacz) [inline backend/unpp_api/apps/project/filters.py]: we don't need comma at the end of list

File: backend/unpp_api/apps/project/filters.py [hunk 2]
```diff
@@ -29,6 +37,11 @@ def get_locations(self, queryset, name, value):
     def get_specializations(self, queryset, name, value):
         return queryset.filter(specializations=value)
 
+    def get_active(self, queryset, name, value):
+        if value:
```
Comment (lkolacz) [inline backend/unpp_api/apps/project/filters.py]: IMO value should be equal to something to dedicated to open like active='true' .. cause it is easier for front-end to set negative value if they touch this field on html once.... here if they set 'false' to expect non active they will get still 'open' project what is wrong.
I'm suggesting to use something like this:
```
    def get_active(self, queryset, name, value):
        if value=='true':
            return queryset.filter(status=EOI_STATUSES.open)
        return queryset.filter(Q(status=EOI_STATUSES.completed) | Q(status=EOI_STATUSES.closed)  # closed is also an information that project is not active IMO
```

File: backend/unpp_api/apps/project/views.py [hunk 3]
```diff
@@ -54,7 +54,8 @@ class BaseProjectAPIView(ListCreateAPIView):
     pagination_class = SmallPagination
     filter_backends = (DjangoFilterBackend, OrderingFilter)
     filter_class = BaseProjectFilter
-    ordering_fields = ('deadline_date', 'start_date', 'status')
+    ordering_fields = ('deadline_date', 'created', 'start_date',
+                       'status', 'completed_date',)
```
Comment (lkolacz) [inline backend/unpp_api/apps/project/views.py]: we need comma at the end of tuple definition only if we have one argument, here it's unnecessary..
btw .. PEP8 says that even if You define comma in tuple there should be white space after comma before "close-parenthesis".

File: backend/unpp_api/apps/project/filters.py (hunk 2 above)
Comment (elyk) [reply backend/unpp_api/apps/project/filters.py]: I added use of BooleanWidget, which will do conversion of all variations of true/false, so will solve this issue. If something sent isn't a variation, it's ignored.

**Pull request merged**